packages/cloud_firestore/cloud_firestore_web/lib/src/interop/firestore.dart

```diff
@@ -140,7 +150,8 @@ class LoadBundleTask
           allowInterop((firestore_interop.LoadBundleTaskProgressJsImpl data) {
         controller.add(LoadBundleTaskProgress._fromJsObject(data));
       }));
-      //todo(russellwheatley): implement error handling for loadBundle
+
+      // todo(russellwheatley): AFAIK - it isn't possible to use catch() as it is a Dart keyword
```

Comment:
No, I tried that but it think it might be deprecated.

Reply:
I was curious about how to work around language mismatches just as a language thing, so I looked - I'm curious if this works in practice as well :nerd_face:

Reply:
Since it is a Promise itself we should just use the [second arg of `then` (`onReject`)](https://firebase.google.com/docs/reference/js/firebase.firestore.LoadBundleTask#then), e.g.:

```dart
  external PromiseJsImpl then([
    void Function(dynamic) onResolve,
    void Function(dynamic) onReject, // <----
  ]);
```